### DISCUSSION TITLE
this.backend undefined during SaveMissing call

### DISCUSSION THREAD

**Discussion by lachauj** [i18next/next-i18next#1921] [General]
I am trying to make it work **saveMissing**.
I added some debug in i18next.js and when saveMissing is called, this.backend is undefined so nothing is sent to i18next-fs-backend.

`if (this.backend && this.backend.create) {
        this.backend.create(languages, namespace, key, fallbackValue, null, _objectSpread$5(_objectSpread$5({}, options), {}, {
          isUpdate: isUpdate
        }));
}`

I noticed that the saveMissing() call is client side and seems to be a problem, should be called on server side right ?

Everything else works perfectly, the existing keys are translated.

**Comment by adrai** [comment]
saveMissing only works if a backend is configured...
for example here it is: https://locize.com/blog/next-i18next/#save-missing (and also only on client side)

**Reply by lachauj** [reply]
Oh but what about the built-in backend i18next-fs-backend ? It will never write any missing keys in the "addPath" ?

**Reply by adrai** [reply]
just tested, seems to work:
<img width="559" alt="image" src="https://user-images.githubusercontent.com/1086194/181442604-6ccca51c-b0ff-4ca5-b22e-f77fa783698f.png">

**Reply by lachauj** [reply]
Can you provide me your boilerplate please ? I've been working on it for 3 days...

**Reply by adrai** [reply]
https://github.com/i18next/next-i18next/tree/master/examples/simple

**Comment by lachauj** [comment]
Thank you, I finally found the issue. In the _app.tsx I had two return.

<img width="225" alt="Screenshot 2022-07-28 at 15 31 09" src="https://user-images.githubusercontent.com/2787672/181561441-57c56832-6e04-429f-a046-bf05feee7e29.png">

Return an empty div for a sec was the issue

Edit: Seems to be an issue for my whole app, missingKeys are detected only on the first return in the component.

**Comment by lachauj** [comment]
<img width="367" alt="Screenshot 2022-07-29 at 12 47 48" src="https://user-images.githubusercontent.com/2787672/181743669-1ee9f8e7-956d-4127-bc98-7c7566c6804d.png">

In this case **loading** is not a state, therefore it will never trigger a re-render on the server side.
So only 'first missing' is detected as a missing key, how can I trigger the detection of the 'second missing' ?

**Reply by adrai** [reply]
I suggest to use saveMissing only on client side
